lib/client-assets.php

```diff
@@ -255,7 +255,7 @@ function gutenberg_register_packages_styles( $styles ) {
 		$styles,
 		'wp-block-editor-content',
 		gutenberg_url( 'build/block-editor/content.css' ),
-		array(),
+		array( 'wp-components' ),
```

Comment:
This enqueues `wp-components` and `dashicons` in the iframe, otherwise they are missing.

`wp-block-editor-content` was introduced at https://github.com/WordPress/gutenberg/pull/44298 as a separate stylesheet for the iframe. It makes sense to add components as a dependency, and I don't see any side effect in non-iframed scenarios (`wp-block-editor` also has components as a dependency).

Reply:
Related to https://github.com/WordPress/gutenberg/pull/49655/files#r1176172481 I'm surprised we have to add these here. `wp-components` is a dependency of `wp-edit-blocks`, which is enqueued in the iframe, hence `wp-components` should as well.

Reply:
`wp-block-editor` uses components from `wp-components`, so we have to add the style dependency as well.

`wp-edit-blocks` should _not_ be added to the iframe, as this loads `block-library/editor.css` (for our core blocks), which is equivalent to `editorStyle` for block.json.

Reply:
> wp-edit-blocks should not be added to the iframe

Ah, nice, thanks for the explanation. `wp-edit-blocks` is present in trunk and in this PR, as far as I tested: [see the table below](https://github.com/WordPress/gutenberg/pull/49655/files#r1176172481).

Are you able to edit my comment there? It'd be great adding a new column sharing what's the end goal, what assets are missing and what assets should not be enqueued. Even if we don't fix it all in this PR, having a reference would be helpful.

Reply:
They are added because of the compat layer. I just checked and we don't actually log a warning, so we should add that in a separate PR. Probably `wp-edit-blocks` is adding some classes targeting blocks, which should also be fixed. Let's do that separately?

Reply:
> They are added because of the compat layer.

Right! This explains it.

Documenting it for our future selves: this is the [compat-layer](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/iframe/use-compatibility-styles.js). It's a mechanism that adds the stylesheets present in the root window that contain the `.wp-block*` CSS selector. It works client-side, which means that it lacks access to the dependency tree of a particular handle. As a result, no dependencies are added to the iframe unless they also contain the `.wp-block*` class (this is unlike adding them in the server, which does consider the dependencies).

In this PR, the stylesheets added using the compat layer are:

- `wp-block-library-css`
- `wp-edit-blocks-css`
- `enqueue_block_editor_assets_style_wp_block-css`

In trunk, the stylesheets added using this mechanism are:

- `enqueue_block_editor_assets_style_wp_block-css`
- `enqueue_block_assets_style_wp_block-css`

Reply:
```suggestion
		array( 'wp-components', 'wp-block-library' ),
```

Reply:
I see. Agreed, enqueue them explicitly makes more sense, actually.